After NX15 upgrade, docker image is not building because of esbuild

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior

After upgrading to NX 15.0, the docker build stopped working.   We are using node:18.11-alpine3.16 on AWS linux. 

<img width="1117" alt="image" src="https://user-images.githubusercontent.com/53616/195994606-c3e77f79-b776-4ccf-9ea4-7b7d2344742c.png">


## Expected Behavior
This is a regression, the only change is upgrading to nx 15. 

## Steps to Reproduce
<!-- Help us help you by making it easy for us to reproduce your issue! -->

<!-- Can you reproduce this on https://github.com/nrwl/nx-examples? -->
<!-- If so, open a PR with your changes and link it below. -->
<!-- If not, please provide a minimal Github repo -->
<!-- At the very least, provide as much detail as possible to help us reproduce the issue -->

<!-- Remove this line -->
This issue may not be prioritized if details are not provided to help us reproduce the issue.

### Failure Logs
<!-- Please include any relevant log snippets or files here. -->

---> Running in affefbe933c6
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for @esbuild/linux-loong64@0.15.5: wanted {"os":"linux","arch":"loong64"} (current: {"os":"linux","arch":"x64"})
npm ERR! notsup Valid OS:    linux
npm ERR! notsup Valid Arch:  loong64
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64

### Environment

KEEP IN MIND, this is a BUILD issue, so the OS is linux.x64, but here is my local.  ESBuild is installed locally only currently.  

Node : 16.16.0
   OS   : darwin arm64
   npm  : 8.11.0

   nx : 15.0.0
   @nrwl/angular : 15.0.0
   @nrwl/cypress : 15.0.0
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.0.0
   @nrwl/esbuild : 15.0.0
   @nrwl/eslint-plugin-nx : 15.0.0
   @nrwl/expo : Not Found
   @nrwl/express : 15.0.0
   @nrwl/jest : 15.0.0
   @nrwl/js : 15.0.0
   @nrwl/linter : 15.0.0
   @nrwl/nest : 15.0.0
   @nrwl/next : Not Found
   @nrwl/node : 15.0.0
   @nrwl/nx-cloud : 14.7.0
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.0.0
   @nrwl/web : Not Found
   @nrwl/webpack : 15.0.0
   @nrwl/workspace : 15.0.0
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
   	 @compodoc/compodoc: 1.1.19
   	 @trumbitta/nx-plugin-unused-deps: 1.12.1

➜  monorepo git:(jbarmash-nx-followup-build-fix) ✗ d

--- a/packages/workspace/src/utilities/create-package-json.ts
+++ b/packages/workspace/src/utilities/create-package-json.ts
@@ -65,10 +65,12 @@ function findAllNpmDeps(
   if (node) {
     list[node.data.packageName] = node.data.version;
     recursivelyCollectPeerDependencies(node.name, graph, list);
+  } else {
+    // we are not interested in the dependencies of external projects
+    graph.dependencies[projectName]?.forEach((dep) => {
+      findAllNpmDeps(dep.target, graph, list, seen);
+    });
   }
-  graph.dependencies[projectName]?.forEach((dep) => {
-    findAllNpmDeps(dep.target, graph, list, seen);
-  });
 
   return list;
 }

